bin/main

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env ruby
+# rubocop:disable Style/ClassVars
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+
+puts '
+_______ _          _______             _______
+|__   __(_)        |__   __|           |__   __|
+   | |   _  ___ ______| | __ _  ___ ______| | ___   ___
+   | |  | |/ __|______| |/ _` |/ __|______| |/ _ \ / _ \
+   | |  | | (__       | | (_| | (__       | | (_) |  __/
+   |_|  |_|\___|      |_|\__,_|\___|      |_|\___/ \___|
+	 '
+
+puts '
+
+
+'
+puts "What's your name?"
+user = gets.chomp
+user = 'Stranger' if user == ''
+puts '
+
+
+'
+puts "Hi #{user}, Welcome to TIC TAC TOE game"
+
+puts '
+
+
+'
+
+# Game instructions
+
+@val = 0
+
+module Display
+  def display_guide
+    puts "
+		"
+    puts '1 | 2 | 3'
+    puts '---------'
+    puts '4 | 5 | 6'
+    puts '---------'
+    puts '7 | 8 | 9'
+    puts "
+		"
+  end
+end
+
+class Board
+  @@squares = {
+    1 => '-',
+    2 => '-',
+    3 => '-',
+    4 => '-',
+    5 => '-',
+    6 => '-',
+    7 => '-',
+    8 => '-',
+    9 => '-'
+  }
+
+  def self.squares
+    @@squares
+  end
+
+  def self.reset
+    @@squares = {
+      1 => '-',
+      2 => '-',
+      3 => '-',
+      4 => '-',
+      5 => '-',
+      6 => '-',
+      7 => '-',
+      8 => '-',
+      9 => '-'
+    }
+  end
+end
+
+class Player
+  include Display
+  attr_reader :name, :letter
+
+  def initialize(name, letter)
+    @name = name
+    @letter = letter
+  end
+
+  def pick_slot
+    loop do
+      board = Board.squares.select { |_key, value| value == '-' }
+      arr1 = []
+      board.each { |k, _v| arr1 << k }
+      display_guide
+      puts "#{@name} choose a number from #{arr1} that represents each slot"
+      @slot = gets.chomp.to_i
+      break if board[@slot] == '-'
+    end
+    @slot
+  end
+end
+
+class Game
+  include Display
+  attr_accessor :val
```

Comment:
Try to use attr_reader otherwise, this exposes your attributes to unknow changes from outside your class.